test/d2l-applied-filters/d2l-applied-filters.html

```diff
@@ -101,6 +101,107 @@
+					const clearFilters = appliedFilters.shadowRoot.querySelector('#d2l-clear-filters-button');
+					const options = dropdown.querySelectorAll('d2l-filter-dropdown-option');
+					options[1].click();
+
+					afterNextRender(options[1], () => {
+						assert.equal(clearFilters.hidden, false);
+						done();
+					});
+				});
+				test('selecting a 4th filter then removing it hides the clear filters button', function(done) {
+					const clearFilters = appliedFilters.shadowRoot.querySelector('#d2l-clear-filters-button');
+					const options = dropdown.querySelectorAll('d2l-filter-dropdown-option');
+					options[1].click();
+
+					afterNextRender(options[1], () => {
+						options[1].deselect();
+
+						afterNextRender(options[1], () => {
+							assert.equal(clearFilters.hidden, true);
+							done();
+						});
+					});
+				});
+				test('clear filters button is visible when all filters are applied', function(done) {
+					const clearFilters = appliedFilters.shadowRoot.querySelector('#d2l-clear-filters-button');
+					const options = dropdown.querySelectorAll('d2l-filter-dropdown-option');
+
+					options[1].click();
+					options[2].click();
+					options[4].click();
+					options[5].click();
+					options[7].click();
+					options[8].click();
+
+					afterNextRender(options[1], () => {
+						assert.equal(clearFilters.hidden, false);
+						done();
+					});
+				});
+				test('clear filters button is hidden when 0 filters are applied', function(done) {
+					const clearFilters = appliedFilters.shadowRoot.querySelector('#d2l-clear-filters-button');
+					const options = dropdown.querySelectorAll('d2l-filter-dropdown-option');
+
+					options[0].deselect();
+					options[3].deselect();
+					options[6].deselect();
+
+					afterNextRender(options[0], () => {
+						assert.equal(clearFilters.hidden, true);
+						done();
+					});
+				});
+				test('clear filters button is hidden when 1 filters is applied', function(done) {
+					const clearFilters = appliedFilters.shadowRoot.querySelector('#d2l-clear-filters-button');
+					const options = dropdown.querySelectorAll('d2l-filter-dropdown-option');
+
+					options[0].deselect();
+					options[3].deselect();
+
+					afterNextRender(options[0], () => {
+						assert.equal(clearFilters.hidden, true);
+						done();
+					});
+				});
+				test('selecting a 4th filter & clicking clear filters fires a d2l-filter-dropdown-option event', function(done) {
+					const clearFilters = appliedFilters.shadowRoot.querySelector('#d2l-clear-filters-button');
+					const options = dropdown.querySelectorAll('d2l-filter-dropdown-option');
+
+					document.addEventListener('d2l-filter-dropdown-cleared', () => {
+						done();
+					});
+
+					options[1].click();
+
+					afterNextRender(clearFilters, () => {
+						clearFilters.click();
+					});
+				});
+				test('selecting a 4th filter & clicking clear filters removes filters & hides clear filters button', function(done) {
```

Comment:
Question, should we actually remove filters in this component, regardless of the filters in the data source? Or should we let the consumer 100% handle it?

Reply:
I hadn't considered letting the consumer handle it themselves. What is the data source for the filters? For us would that be the variables in folio-app that hold the list of applied filters?

Reply:
The data source for the filters is essentially the dropdown. When you send the event in folio, I bet that it'll clean up this component without any code in here doing so (worth a try!)

The reason for doing it that way is otherwise the dropdown and the applied filters component could potentially get out-of-sync, which could be confusing for the user

Reply:
There is code (that I didn't write) in d2l-applied-filters.js that clears the filters in the applied filters component when the `d2l-filter-dropdown-cleared` event is fired
```
target.addEventListener('d2l-filter-dropdown-cleared', this._clearSelected, true);
```
What my code was doing (until I changed it a few minutes ago) was calling `this._clearSelected` to clear the applied filters component when clear filters is pressed, but it was doing that right before firing the `d2l-filter-dropdown-cleared` so I removed my redundant function call

Reply:
If we change it so the consumer handles the clearing of the applied filters component, we would have to use a different event than `d2l-filter-dropdown-cleared`

Reply:
Interestinggg, what's here is likely good then :+1: